### PR TITLE
Compatibility for solc 0.5/0.6

### PR DESCRIPTION
**Pull request by gbalabasquer**
This PR proposes to remove `DSTokenBase` and `ERC20` contracts to be able to compile `DSToken` with solc 0.5 and 0.6.

It doesn't touch anything regarding `symbol` and `name` which are `bytes32` in opposition to the standard `string`. That can be discussed separately.

### CODE REVIEW

**Comment by gbalabasquer** [comment]
FV specs fully being accepted with solc 0.5.12 - non optimized:
https://github.com/makerdao/k-ds-token

(In order to simplify FV, `DSStop` has been removed as a submodule and the same functionality incorporated directly in the token code with a simpler event emission).

**Comment by gbalabasquer** [comment]
FV with 0.6.7 - non optimized is also fully being accepted.

**Comment by gbalabasquer** [comment]
FV with 0.6.7 being fully accepted with optimization and 999999 runs: https://github.com/makerdao/k-ds-token/tree/32963ef8c0c93906e1b9f73b976d2c4edbdc36be

Report: https://reports.makerfoundation.com/k-ds-token/8ef623b0b787d7777c8d/

cc @xwvvvvwx

File: src/token.sol [hunk 1]
```diff
@@ -17,86 +17,120 @@
...
+    event Start();
+
+    modifier stoppable {
+        require(!stopped, "ds-stop-is-stopped");
+        _;
+    }
 
-    function approve(address guy) public stoppable returns (bool) {
-        return super.approve(guy, uint(-1));
+    function approve(address guy) external returns (bool) {
```
Comment (xwvvvvwx) [inline src/token.sol:49]: why not keep this as `stoppable`?

Comment (MrChico) [reply src/token.sol:49]: the `approve` that it calls is stoppable anyway. But yeah, making it explicit is nicer I suppose

Comment (xwvvvvwx) [reply src/token.sol:49]: ah of course, missed that :woman_facepalming: 

I'm fine with leaving this as it is then, it's actually more consistent wtih the other methods that call into `stoppable` functions (`push`, `pull`, `move`, etc).

**Pull request merged**